### PR TITLE
Resolve child resource error and other minor bugs

### PR DESCRIPTION
**Pull request by sapsh-msft**
## Description
</br>
 Brief description of the requested change.

## Checklist
- [ ] I have read the instructions mentioned in [Contribute to Code](/Contributing.md).
- [ ] I have read and understood the criteria described under [submitting changes](/Contributing.md#submitting-changes). 
- [ ] The title of the PR clearly describes the intent of the PR.
- [ ] This PR does not introduce any breaking changes to the code.

### CODE REVIEW

File: src/AzSK/Framework/Core/AzSKInfo/ComplianceInfo.ps1 [hunk 1]
```diff
@@ -293,7 +293,7 @@ class ComplianceInfo: CommandBase
 					{
 						$passControlCount++
 						#baseline controls condition shouldnot increment if it wont fall in passed/ failed state
-						if($_.IsBaselineControl)
+						if($_.IsBaselineControl.ToLower() -eq "true")
```
Comment (VishalHaibatpure) [inline src/AzSK/Framework/Core/AzSKInfo/ComplianceInfo.ps1]: Powershell auto type case to boolean value. Do we really need this?

Comment (sapsh-msft) [reply src/AzSK/Framework/Core/AzSKInfo/ComplianceInfo.ps1]: $_.IsBaselineControl contains string value so no matter value is true or false if block will return true.
So either we need to cast the string to bool and compare or string comparison.

Comment (VishalHaibatpure) [reply src/AzSK/Framework/Core/AzSKInfo/ComplianceInfo.ps1]: Make IsBaselneControl as bool in class ComplianceResult

Comment (sapsh-msft) [reply src/AzSK/Framework/Core/AzSKInfo/ComplianceInfo.ps1]: The complianceresult class is used to print data in CSV. So if we take value in bool previously in last we need to convert it to string as "Yes" or "No" in other property. So we have taken this property as string.

**Pull request merged**